Modules/_datetimemodule.c

```diff
@@ -2284,21 +2284,69 @@ delta_bool(PyDateTime_Delta *self)
 static PyObject *
 delta_repr(PyDateTime_Delta *self)
 {
-    if (GET_TD_MICROSECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self),
-                                    GET_TD_MICROSECONDS(self));
-    if (GET_TD_SECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self));
-
-    return PyUnicode_FromFormat("%s(%d)",
-                                Py_TYPE(self)->tp_name,
-                                GET_TD_DAYS(self));
+    PyObject *args = PyList_New(0);
+
+    if (args == NULL) {
+        return NULL;
+    }
+
+    if (GET_TD_DAYS(self) != 0) {
+        PyObject *days = PyUnicode_FromFormat("days=%d", GET_TD_DAYS(self));
+        if (days == NULL || PyList_Append(args, days) < 0) {
+            Py_XDECREF(days);
+            goto error;
+        }
+        Py_DECREF(days);
+    }
+
+    if (GET_TD_SECONDS(self) != 0) {
+        PyObject *seconds = PyUnicode_FromFormat("seconds=%d",
+                                                 GET_TD_SECONDS(self));
+        if (seconds == NULL || PyList_Append(args, seconds) < 0) {
+            Py_XDECREF(seconds);
+            goto error;
+        }
+        Py_DECREF(seconds);
+    }
+
+    if (GET_TD_MICROSECONDS(self) != 0) {
+        PyObject *microseconds = PyUnicode_FromFormat("microseconds=%d",
+                                                      GET_TD_MICROSECONDS(self));
+        if (microseconds == NULL || PyList_Append(args, microseconds) < 0) {
+            Py_XDECREF(microseconds);
+            goto error;
+        }
+        Py_DECREF(microseconds);
+    }
+
+    if (PyList_Size(args) == 0) {
+        Py_DECREF(args);
+        return PyUnicode_FromFormat("%s(0)", Py_TYPE(self)->tp_name);
+    }
+
+    PyObject *sep = PyUnicode_FromString(", ");
+    if (sep == NULL) {
+        Py_DECREF(args);
+        return NULL;
+    }
+
+    PyObject *args_string = PyUnicode_Join(sep, args);
+    Py_DECREF(sep);
+    Py_DECREF(args);
+
+    if (args_string == NULL) {
+        return NULL;
+    }
+
+    PyObject *repr = PyUnicode_FromFormat("%s(%S)",
+                                          Py_TYPE(self)->tp_name,
```

Comment:
Yes, it uses the fully qualified name. Otherwise, the tests would have caught it. I did reply to it, but Github probably collapsed the comment.